### ISSUE TITLE
Error in the alias API [CL integration]

### ISSUE THREAD

**Issue by nirmalpatel** [CarnegieLearningWeb/UpGrade#59]
Mike F reported several errors related to alias API.

```
post /api/useraliases
{"type":"Query Failed","message":"Users with ids mx-2ac29136-68f4-4fff-8b86-256b63e66b72-modeling_linear_equations,mx-2ac29136-68f4-4fff-8b86-256b63e66b72-developing_functional_foundations already associated with other users"}
```

We need to fix this and add additional logging data in this error message as mentioned in #58.

**Comment by VivekFitkariwala** [comment]
Created a [PR](https://github.com/CarnegieLearningWeb/UpGrade/pull/60) with additional error logs.

Two scenarios should not be created from client application
- An alias pointing to a user should not be updated to point another user.
- An existing user (which is not an alias) should be used alias of other user.

**Comment by nirmalpatel** [comment]
We have two types of users in UpGrade. Root users that get enrolled in experiments, and alias users that can point to one and only one root user. A root user can have multiple aliases.

There are multiple scenarios of erroneous behavior when it comes to aliases. Now we have identified two distinct scenarios and have added detailed error messages for each of them:

1. An alias ID is already associated with another root ID
2. A root ID is trying to become an alias of another root ID

Both of the above cases now lead to an error and give more details about the user and aliases.

**Comment by nirmalpatel** [comment]
@amurphy-cl @msandbothe Vivek merged PR #60 and that closed issue #58. I'm wondering what series of calls are leading to this error now... did the new error messages help @msandbothe to diagnose the issue?

**Comment by msandbothe** [comment]
Continued by #69
